Geometry_Engine/Compute/LargestComponentVector.cs

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Base;
+using BH.oM.Geometry;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****          Public Methods                   ****/
+        /***************************************************/
+
+        [Description("Returns the largest component as determined by two provided points, dependent upon which axis is more represented by the difference between the two points.")]
+        [Input("startPoint", "The start point of vector.")]
+        [Input("endPoint", "The end point of the vector.")]
+        [Output("largestComponentVector", "The largest component vector as determined by two provided points.")]
+        public static Vector LargestComponentVector(Point startPoint, Point endPoint)
+        {
+            //If points are equal, no largest component
+            if (startPoint == endPoint)
```

Comment:
Wrap this in brackets as it's going over multiple lines in the `return` - just the make code easier to read